jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainFragment.java

```diff
@@ -184,6 +174,44 @@ public void onClick(View view) {
         });
     }
 
+    /** Sends the user entered message if permission was granted. Requests the permission if the app
+     * does not already have it. Also shows a dialog if now message was entered. */
+    private void sendMessage() {
+
+        message = editMessage.getText().toString();
+
+        if (message.length() <= 0) {
+            EnterMessageDialogFragment dialog = new EnterMessageDialogFragment();
+            dialog.show(getFragmentManager(), ENTER_MESSAGE_FRAGMENT_TAG);
+            return;
+        }
+
+        // If the user does not wish to send an SMS.
+        if (contactPhoneNumber == null) {
+            return;
+        }
+
+        delegate.sendMessage(message);
+        editMessage.setText("");
```

Comment:
Misplaced this somehow - moved it up before the check above. The point was that the use could still send a message to himself (like in the previous versions) if no number was entered.